openedx/core/djangoapps/user_authn/views/register.py

```diff
@@ -170,9 +167,14 @@ def create_account_with_params(request, params):
     third_party_auth_credentials_in_api = 'provider' in params
     is_third_party_auth_enabled = third_party_auth.is_enabled()
 
-    if is_third_party_auth_enabled and (pipeline.running(request) or third_party_auth_credentials_in_api):
+    is_sso = is_sso_request(request)
+    if is_sso:
         params["password"] = generate_password()
 
+    if is_registration_api_v1(request) or is_sso:
```

Comment:
Since `third_party_auth_credentials_in_api` essentially means the `request` is made from the mobile application, so this new check `is_sso` will impact mobile as well now?

Reply:
changed `is_sso` logic and moved `third_party_auth_credentials_in_api` out of it.